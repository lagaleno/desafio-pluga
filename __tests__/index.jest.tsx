// Jest imports
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import 'intersection-observer';

// Component import
import Home from './../src/pages'

// Mock import
import toolsMock from "../__mocks__/toolsMock.json";

describe ('Home', () => {
    it( 'Renders components correctly', () => {
        const { container } = render (<Home tools={toolsMock} />)
        expect(container).toMatchSnapshot()
    })
})