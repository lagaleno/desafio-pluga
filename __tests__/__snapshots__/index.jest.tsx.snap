// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Renders components correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-1c5ij41"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-i3xwwb-MuiGrid-root"
      >
        <form>
          <div
            class="sc-jTQDnj cQbUnX"
          >
            <button
              aria-label="search"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="submit"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SearchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="search-bar"
                id="search-bar-label"
              >
                Entre com o nome da Ferramenta
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="search-bar"
                  placeholder="Buscar Ferramenta"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Entre com o nome da Ferramenta
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </form>
        <div
          style="padding: 3px;"
        />
      </div>
    </div>
    <div
      class="MuiBox-root css-vje9uk"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-i46k62-MuiGrid-root"
      >
        <div
          style="width: 100%; display: flex; align-items: center;"
        >
          <div
            class="sc-eDLJxc jPXNHE"
          >
            <span
              class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
              role="progressbar"
              style="width: 40px; height: 40px;"
            >
              <svg
                class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                  cx="44"
                  cy="44"
                  fill="none"
                  r="20.2"
                  stroke-width="3.6"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-1c5ij41"
    >
      <div
        class="sc-gLLvby hubxQF"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <span
              class="MuiTypography-root MuiTypography-overline MuiTypography-alignRight css-1hajmvu-MuiTypography-root"
            >
              Feito por: 
              <a
                href="https://github.com/lagaleno"
              >
                 @lagaleno 
              </a>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
